### ISSUE TITLE
[BUG] add-packages to `dependencies:` section.

### ISSUE THREAD

**Issue by mforbes** [Anaconda-Platform/anaconda-project#337]
When an `anaconda-project.yml` file has a `dependencies:` section rather than a `packages:` section as allowed by the alias implemented in PR #200 and requested in issue #194 so that `anaconda-project.yml` files can act as `environment.yml` files, new packages should be added to the existing section.

What happens instead is that `anaconda-project add-packages` inserts and additional `packages:` section with the new package rather than reusing the existing `dependencies:` section.  Thus, `anaconda-project add-packages pandas` migh produce something like this:

```
#anaconda-project.yaml
...
dependencies:
  - numpy
  - matplotlib
packages:
  - pandas
```

when it should produce something like:

```
#anaconda-project.yaml
...
dependencies:
  - numpy
  - matplotlib
  - pandas
```

**Comment by AlbertDeFusco** [comment]
ah! thanks for that.

**Comment by mforbes** [comment]
I will try to take a stab at fixing this.  

As per @jbednar's request in #194, can we replace `packages` with `dependencies` as the recommended approach going forward once this is resolved?  I don't see any good reason for recommending `packages` except for this issue.

**Comment by AlbertDeFusco** [comment]
I agree. I've been looking at the code and I see that `packages` is somewhat hardcoded. I'm trying to come up with searches to help understand

```
❯ grep "\['packages'\]" -r .
./anaconda_project/env_spec.py:        template_json['something']['packages'] = packages
./anaconda_project/test/test_project_ops.py:            assert spec in env_commented_map['packages'], env_commented_map['packages']
./anaconda_project/project_file.py:                move_list_elements(spec_json['packages'], as_json['packages'])
./anaconda_project/conda_manager.py:        yaml_dict['packages'] = packages_dict
./anaconda_project/project_ops.py:        env_dict['packages'] = old_packages
./anaconda_project/project_ops.py:                        env_dict['packages'].append({'pip': []})
./anaconda_project/project_ops.py:                        env_dict['packages'] = [{'pip': []}]
./anaconda_project/project_ops.py:                    env_dict['packages'].append({'pip': pip_packages})
./anaconda_project/project_ops.py:                env_dict['packages'] = old_packages
./anaconda_project/project_ops.py:            env_dict['packages'] = old_packages
./anaconda_project/requirements_registry/providers/test/test_conda_env.py:        project.project_file.set_value(['packages', deps + ['boguspackage'])
```

```
❯ grep -E "get\('packages'.*\)" -r .
./anaconda_project/env_spec.py:        raw_dependencies = yaml.get('packages', [])
./anaconda_project/project_ops.py:        old_packages = env_dict.get('packages', [])
./anaconda_project/project_ops.py:            _pkgs = env_dict.get('packages', [])
./anaconda_project/project_ops.py:            old_packages = env_dict.get('packages', [])
./anaconda_project/project_ops.py:            old_packages = env_dict.get('packages', [])
./anaconda_project/project.py:            packages_by_platform = lock_set.get('packages', {})
./anaconda_project/project.py:                    packages = env_dict.get('packages', [])
```

This seems like a good opportunity to adopt black (or some other linter/formatter) to swap all singe to double quotes to help my grepping

**Comment by mforbes** [comment]
@AlbertDeFusco I will push some code later today, but I have made a test fixture that should run all tests with `'dependencies'` and `'packages'`.  I will try to fix some of the resulting errors, but will probably need your guidance tomorrow (did you get my meeting request).

**Comment by AlbertDeFusco** [comment]
Great! Looking forward to meeting tomorrow

**Comment by mcg1969** [comment]
Thanks for opening this issue, and apologies for the long delay in responding.

Anaconda Project has not been actively developed in some time and has largely been superseded by newer alternatives:

* [pixi](https://pixi.sh/) and [uv](https://docs.astral.sh/uv/) represent the first truly successful successors.
* [conda-workspaces](https://github.com/conda-incubator/conda-workspaces) is a promising pixi-compatible alternative for users who prefer to stay closer to the conda ecosystem.

Going forward, work on this repository is focused on critical bug fixes and on tooling to help users migrate existing Anaconda Project projects to pixi (and, by extension, conda-workspaces). See the [Pixi support](https://anaconda-project.readthedocs.io/en/latest/pixi-support.html) page in our documentation for details on the `anaconda-project export-pixi` command, which can convert an existing `anaconda-project.yml` into a `pixi.toml` for you.

The `anaconda-project` package itself will remain available, so existing projects relying on its stable feature set can continue to be supported. However, we do not anticipate further enhancement work, and for that reason we are closing out the bulk of our open issue backlog — including this one.

If this issue describes a critical regression in the stable feature set (as opposed to an enhancement or an edge case in a feature we no longer plan to expand) and you have a minimal reproducer against a current release, please feel free to reopen and we will take another look.

Thank you for your interest in Anaconda Project.

**Comment by mcg1969** [comment]
Reopening — on closer review during a re-audit pass, this is a real bug that's cheap to fix correctly. PR #433 picks the existing key (`dependencies` if present, else `packages`) at every mutation site in `add_packages`/`remove_packages` and writes back to it. This issue will auto-close when #433 merges.
